src/codec/PartyDetailsCodec.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import {Codec} from "./Codec";
+import {PartyDetails} from "../model/PartyDetails";
+import {PartyDetails as PbPartyDetails} from "../generated/com/digitalasset/ledger/api/v1/admin/party_management_service_pb"
+
+export const PartyDetailsCodec: Codec<PbPartyDetails, PartyDetails> = {
+    deserialize(request: PbPartyDetails): PartyDetails{
+        return {
+            party: request.getParty(),
+            display_name: request.getDisplayName(),
```

Comment:
Make sure to set `displayName` only if the value is actually there. Other codecs may give you a hint on how to treat optional fields. In particular, the [`CompletionCodec`](https://github.com/digital-asset/daml-js/blob/fd3bea9766858e84d779aef823cb621596c8441a/src/codec/CompletionCodec.ts) shows an example of how to treat a native string which happens to be a native string.